libfprint/drivers/goodixtls/5395/goodix_device.c

```diff
@@ -258,3 +268,94 @@ gboolean fpi_goodix_device_reset(FpDevice *dev, guint8 reset_type, gboolean irq_
     GError *error = NULL;
     return fpi_goodix_device_send(dev, message, TRUE, 500, FALSE, &error);
 }
+
+gboolean fpi_goodix_device_gtls_connection(FpDevice *dev, GError *error) {
+    fp_dbg("fpi_goodix_device_gtls_connection()");
+    fp_dbg("Starting GTLS handshake");
+    FpiGoodixDevice *self = FPI_GOODIX_DEVICE(dev);
+    FpiGoodixDevicePrivate *priv = fpi_goodix_device_get_instance_private(self);
+    priv->gtls_params = fpi_goodix_device_gtls_init_params();
+
+    // Hello phase
+    fpi_goodix_gtls_create_hello_message(priv->gtls_params);
```

Comment:
maybe better return hello message from the function and set it here in priv->gtls_params->client_random